src/classes/CMT_MetadataAPI.cls

```diff
@@ -168,26 +165,78 @@ public class CMT_MetadataAPI {
      * any error messages returned by the api call.
      */
     public class MetadataCallBack implements Metadata.DeployCallBack {
+
+        private String jobId;
+        private List<List<Metadata.CustomMetadata>> nextOperation;
+
+        /**
+         * @description Constructor for Callback Class
+         * @param jobId Some unique string value to use when writing the deployment status to the Custom Settings object
+         * @param nextOperation List<CustomMetadata> for next operation if there are any child records to deploy.
+         */
+        public MetadataCallBack(String jobId, List<List<Metadata.CustomMetadata>> nextOperation) {
+            this.jobId = jobId;
+            this.nextOperation = nextOperation;
+        }
+
+        /**
+         * @description Method called by salesforce with the result of the Metadata deployment operation status.
+         * This will either write the status to a CustomSettings object, or queue up the next chained deployment operation.
+         */
         public void handleResult(Metadata.DeployResult result, Metadata.DeployCallbackContext context) {
+
+            Boolean isSuccess = (result != null ? result.status == Metadata.DeployStatus.Succeeded : false);
+            Map<String, String> statusMap = new Map<String, String>{ this.jobId => (result != null ? result.status.name() : 'Unknown') };
+            // If the staus is not Success, collect all errors into a string that can be written as the status
             if (result != null && result.status != Metadata.DeployStatus.Succeeded) {
                 String errMsg = '';
                 for (Integer n=0; n < result.details.componentFailures.size(); n++) {
-                    if (n > 0) {
-                        errMsg += '\n';
-                    }
-                    errMsg += '* ' + result.details.componentFailures[n].fullName + ': ' + result.details.componentFailures[n].problem;
+                    errMsg += (n > 0 ? '\n' : '') + '* ' + result.details.componentFailures[n].fullName +
+                            ': ' + result.details.componentFailures[n].problem;
+                }
+                // The custom settings field is a max of 255 characters, but need space for the serialized json overhead
+                statusMap.put(this.jobId, errMsg.left(220));
+                isSuccess = false;
+            }
+
+            // If there are more records to deploy, chain the Queuable class to continue the deployment
+            if (isSuccess && this.nextOperation != null && this.nextOperation.size() > 0 && !Test.isRunningTest()) {
+                System.enqueueJob(new QueueableDeployMetadata(this.jobId, this.nextOperation));
+
+            } else {
+                // Otherwise, write the deployment status to the custom settings object
```

Comment:
Unfortunately their is a scenario where Apex fails with a fatal error that never actually calls the callback class. It would be better if the caller that implements this method handles the retry since it can use a javascript/page based polling process that can look for an updated status and retry if none is returned in a specified amount of time.